pkg/globalnet/controllers/ipam/ipam.go

```diff
@@ -124,6 +147,34 @@ func (i *Controller) processNextObject(objWorkqueue workqueue.RateLimitingInterf
 				}
 				return fmt.Errorf("error retrieving submariner-ipam-controller object %s: %v", name, err)
 			}
+
+			switch runtimeObj.(type) {
+			case *k8sv1.Service:
+				service := runtimeObj.(*k8sv1.Service)
+				switch i.processServiceStatus(service) {
+				case Ignore:
+					objWorkqueue.Forget(obj)
+					return nil
+				case Requeue:
+					objWorkqueue.AddRateLimited(obj)
+					return fmt.Errorf("service %s requeued %d times", key, objWorkqueue.NumRequeues(obj))
+				}
+			case *k8sv1.Pod:
+				pod := runtimeObj.(*k8sv1.Pod)
+				// Process pod event only when it has an ipaddress. Pods skipped here will be handled subsequently
+				// during podUpdate event when an ipaddress is assigned to it.
+				if pod.Status.PodIP == "" {
+					objWorkqueue.Forget(obj)
+					return nil
+				}
+
+				// Privileged pods that use hostNetwork will be ignored.
+				if pod.Status.PodIP != "" && pod.Status.PodIP == pod.Status.HostIP {
```

Comment:
nit: no need to check `pod.Status.PodIP != ""` here since that's done above.

Reply:
Done